platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/properties/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  @Test
+  public void testAttemptLoadWithSpring() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpring(
+            PROPERTIES_FILENAME, this.getClass().getClassLoader());
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpringFileDNE() throws Exception {
+    Properties testProperties = PropertiesLoader.attemptLoadWithSpring(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpringAndClassLoader() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpringAndClassLoader(
+            PROPERTIES_FILENAME, this.getClass().getClassLoader());
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpringAndClassLoaderFileDNE() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpringAndClassLoader(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithFileSystem() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithFileSystem(propertiesFile.getPath(), null);
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithFileSystemFileDNE() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithFileSystem(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadAsResource() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadAsResource("/" + PROPERTIES_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadAsResourceFileDNE() throws Exception {
+    Properties testProperties = PropertiesLoader.attemptLoadAsResource(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testSubstituteSystemPropertyPlaceholders() throws Exception {
+    // set up map to replace system properties
+    Map<Object, Object> testMapSystemProperties = new HashMap<>();
+    testMapSystemProperties.put("Java Home", "${java.home}");
+    testMapSystemProperties.put("Java Version", "${java.version}");
```

Comment:
❗️ We probably should not have spaces in the keys. They are usually of the form `org.codice.qualifier.qualifier.propertyname` and we should test as such. Also we should test for system property expansions in the keys as well.